### PR TITLE
Fix helpful error

### LINKED ISSUE #1365: bad error message in variable
```julia
julia> m = Model()
A JuMP Model

julia> @variable(m, 0 <= x)
ERROR: UndefVarError: S not defined
```

### PR DESCRIPTION
**Pull request by blegat**
Closes #1365

### CODE REVIEW

**Comment by odow** [comment]
Should we throw a proper exception type to distinguish the helpful error message from the un-helpful "S not defined"? Because this wasn't caught in tests as it still throws an error.

**Comment by mlubin** [comment]
Can't we just make this work? 
```julia
if var isa Number && !(value isa Number)
     parse_one_operator_variable(_error, infoexpr, Val(reverse_sense(sense)), esc_nonconstant(var))
    return value
end
```

**Comment by blegat** [comment]
I don't understand why it wasn't called since we check that its an `ErrorException` and `S not defined` should be an `UndefVarError`.

**Comment by odow** [comment]
The question is whether it is more or less confusing for users. Then you have to say why
```julia
@variable(m, 0 <= x)
```
works but
```julia
a = 0
@variable(m, a <= x)
```
I can see it leading to more confusion than "it never works"

**Comment by blegat** [comment]
> Can't we just make this work?

We should at least throw a warning because it's going to fail badly when it's not constant.
Or we add in the FAQ the meaning of the weird error that is got. I am fine with both.

**Comment by mlubin** [comment]
The confusion always seems to be with
```julia
@variable(m, 0 <= x)
```
I've yet to see a complaint about
```julia
a = 0
@variable(m, a <= x)
```

**Comment by odow** [comment]
Okay lets change it.

**Comment by blegat** [comment]
I have changed it. It now works instead of erroring :)

File: src/macros.jl [hunk 1]
```diff
@@ -965,18 +965,22 @@ function parsevariable(_error::Function, infoexpr::VariableInfoExpr, sense::Symb
     # Variable declaration of the form: var sense value
```
Comment (mlubin) [inline src/macros.jl]: This comment is out of date, so are the argument names.

Comment (blegat) [reply src/macros.jl]: I explains where the function arguments come from. We should probably rename to lhs rhs though

Comment (blegat) [reply src/macros.jl]: I have refactored it in two methods

File: test/variable.jl [hunk 2]
```diff
@@ -34,26 +34,35 @@ function variables_test(ModelType::Type{<:JuMP.AbstractModel}, VariableRefType::
...
-            @test !JuMP.haslowerbound(ubonly)
-            @test JuMP.hasupperbound(ubonly)
-            @test JuMP.upperbound(ubonly) == 1.0
-            @test !JuMP.isfixed(ubonly)
-            @test !JuMP.isbinary(ubonly)
-            @test JuMP.isinteger(ubonly)
-            @test isequal(mcon[:ubonly],ubonly)
+            @variable(mcon, ubonly1 <= 1, Int)
+            @test isequal(mcon[:ubonly1],ubonly1)
+            @variable(mcon, 1 >= ubonly2, Int)
```
Comment (mlubin) [inline test/variable.jl]: Make a separate test for this.

**Pull request merged**